Add function `retagByK8sMeta` and opt type `K8sRetagType.Pod2Service` in MAL for k8s to relate pods and services. (#6608)
apache/skywalking: docs/en/concepts-and-designs/mal.md

```
instance_trace_count.valueGreaterEqual(33)
```
### K8s
MAL support add specific Labels to the samples that collect from K8s metrics collectors.
This feature need OAP Server has the authority to access the K8s's `API Server`.

#### k8sTagServiceByPodName

> `k8sTagServiceByPodName(podName,serviceName)`. Add a sevice Label to the sample that already has a pod name.

Reviewer: I think base on your description, this should be renamed to 
```suggestion
`relabelByK8sMeta(newLabelName, type, existingLabelName)`. Add a new label to the sample family based on an existing label's value. Provide several internal converting types, including
- Pod2Service
```

Reviewer: Does this look better?

Author: done

For example:
```
container_cpu_usage_seconds_total{container=my-nginx, cpu=total, pod=my-nginx-5dc4865748-mbczh} 2
```
Expression:
```
container_cpu_usage_seconds_total.k8sTagServiceByPodName('pod' , 'service')
```
Output:
```

> container_cpu_usage_seconds_total{container=my-nginx, cpu=total, pod=my-nginx-5dc4865748-mbczh, service='default:nginx-service'} 2

Reviewer 2: *svc*.*namespace* is the canonical format for a service, for example, `nginx-service.default`

Author: done

```
instance_trace_count.valueGreaterEqual(33)
```
### K8s
MAL support add specific Labels to the samples that collect from K8s metrics collectors.
This feature need OAP Server has the authority to access the K8s's `API Server`.
#### retagByK8sMeta
`retagByK8sMeta(newLabelName, K8sRetagType, existingLabelName)`. Add a new label to the sample family based on an existing label's value. Provide several internal converting types, including

> - K8sRetagType.Pod2Service  

Reviewer: ```suggestion
- K8sRetagType.Pod2Service  

```

```
instance_trace_count.valueGreaterEqual(33)
```

> ### K8s

Reviewer: ```suggestion
### Tag manipulator
#### K8s
```

> MAL support add specific Labels to the samples that collect from K8s metrics collectors.
> This feature need OAP Server has the authority to access the K8s's `API Server`.

Reviewer: ```suggestion
MAL supports using the metadata of k8s to manipulate the tags and their values.
This feature requires OAP Server to have the authority to access the K8s's `API Server`.
```

#### retagByK8sMeta

> `retagByK8sMeta(newLabelName, K8sRetagType, existingLabelName)`. Add a new label to the sample family based on an existing label's value. Provide several internal converting types, including

Reviewer: ```suggestion
`retagByK8sMeta(newLabelName, K8sRetagType, existingLabelName)`. Add a new tag to the sample family based on an existing label's value. Provide several internal converting types, including
```

- K8sRetagType.Pod2Service  

> Add a sevice Label to the sample that already has a pod name.

Reviewer: ```suggestion
Add a tag to the sample by using `service` as the key, `serviceName.namespace` as the value, by the given value of the tag key, which represents the name of a pod.
```

> The newLabelValue is `serviceName.namespace`.

Reviewer: ```suggestion
```

```
instance_trace_count.valueGreaterEqual(33)
```
### Tag manipulator
MAL provides tag manipulators to change(add/delete/update) tags and their values.

#### K8s
MAL supports using the metadata of k8s to manipulate the tags and their values.
This feature requires OAP Server to have the authority to access the K8s's `API Server`.

> #### retagByK8sMeta

Reviewer: ```suggestion

##### retagByK8sMeta
```
